src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1665,13 +1674,13 @@ let rec TryTranslateComputationExpression
                 | None ->
                     // "do! expr; cexpr" is treated as { let! () = expr in cexpr }
                     match innerComp1 with
-                    | SynExpr.DoBang(expr = rhsExpr; range = m) ->
+                    | SynExpr.DoBang(expr = rhsExpr; trivia = { DoBangKeyword = mKeyword }; range = m) ->
                         let sp =
                             match sp with
                             | DebugPointAtSequential.SuppressExpr -> DebugPointAtBinding.NoneAtDo
                             | DebugPointAtSequential.SuppressBoth -> DebugPointAtBinding.NoneAtDo
-                            | DebugPointAtSequential.SuppressStmt -> DebugPointAtBinding.Yes m
-                            | DebugPointAtSequential.SuppressNeither -> DebugPointAtBinding.Yes m
+                            | DebugPointAtSequential.SuppressStmt -> DebugPointAtBinding.Yes mKeyword
+                            | DebugPointAtSequential.SuppressNeither -> DebugPointAtBinding.Yes mKeyword
```

Comment:
When `do!` in a sequential expression is converted to a `SynBinding`, we were using the whole expression range instead of the keyword range for `SynLeadingKeyword.LetBang`. This caused the error to be reported twice with different ranges (keyword vs expression), so deduplication didn't apply.

Using `DoBangKeyword` ensures both error reports use the same range and get deduplicated.

Fixes the failing tests:
- `do! expressions may not be used in queries`
- `do! expressions may not be used in queries(SynExpr.Sequential)`